### ISSUE TITLE
CRL -  crl with "No Revoked Certificates" 

### ISSUE THREAD

**Issue by mprevide** [vernemq/vernemq#1337]
When I try to use a CRL with "No Revoked Certificates" , this error occurs: 
Logs:

```
17:07:28.970 [error] gen_server vmq_crl_srv terminated with reason: no function clause matching vmq_crl_srv:'-handle_call/3-lc$^1/1-1-'(asn1_NOVALUE) line 84

17:07:28.970 [error] CRASH REPORT Process vmq_crl_srv with 0 neighbours crashed with reason: no function clause matching vmq_crl_srv:'-handle_call/3-lc$^1/1-1-'(asn1_NOVALUE) line 84

17:07:28.971 [error] Supervisor vmq_server_sup had child vmq_crl_srv started with vmq_crl_srv:start_link() at <0.559.0> exit with reason no function clause matching vmq_crl_srv:'-handle_call/3-lc$^1/1-1-'(asn1_NOVALUE) line 84 in context child_terminated

17:07:28.971 [info] TLS server: In state certify at tls_connection.erl:1002 generated SERVER ALERT: Fatal - Handshake Failure - malformed_handshake_data
```

Line:
https://github.com/vernemq/vernemq/blob/5d5a16737e39b321180c4361c8510e291d99565f/apps/vmq_server/src/vmq_crl_srv.erl#L84

Ps.: When my CRL has one or more "Revoked Certificates" works fine.

**Comment by ioolkos** [comment]
@mprevide thanks for opening the issue!
I don't have an answer/understanding immediately, but we'll look into it.

**Comment by larshesel** [comment]
HI, I was trying to reproduce this but wasn't able to so far. I tried with a completely empty crl.pem, one with just the PEM headers, but without any luck. Can you provide a bit more information? Which VerneMQ version, which Erlang/OTP version and perhaps an example crl.pem file?

**Comment by mprevide** [comment]
I'm using erlio/docker-vernemq:1.9.2-alpine.

CRLs PEM:
https://gist.github.com/mprevide/49c43ea3133c3781d69d79b864aa8ebf

Thank you :smile_cat:

**Comment by mprevide** [comment]
Is that enough?

**Comment by larshesel** [comment]
Hi, I weren't able to reproduce this with the above crls - I'm afraid it might be my limited understanding of how CRLs work - can you describe your setup and how and under what circumstances this happen?

**Comment by larshesel** [comment]
I was able to trigger it by manually (by simply feeding the crl file directly to the code) and created a bugfix to handle it in #1341

**Comment by mprevide** [comment]
Hi 
When I try publish this happens. 

I set up a key, a CA certificate, a broker certificate, and the CRL.
When I set up a CRL without any revoked certificates, no matter what certificate I use to publish, this error happens.

This error happens when I publish using the key, CA certificate, and device certificate.
`mosquitto_pub -h 172.17.0.2 -p 8883 -t /testTopic -m '{"a": 1}' -d  --ce
rt admin_d885fd.crt  --key admin_d885fd.key --cafile ca.crt `

**Comment by mprevide** [comment]
> Hi, I weren't able to reproduce this with the above crls - I'm afraid it might be my limited understanding of how CRLs work - can you describe your setup and how and under what circumstances this happen?

Hi 

I set up a key, a CA certificate, a broker certificate, and the CRL in VerneMQ.
When I set up a CRL without any revoked certificates, no matter what certificate I use to publish, this error happens.

This error happens when I publish using the key, CA certificate, and device certificate.
`mosquitto_pub -h 172.17.0.2 -p 8883 -t /testTopic -m '{"a": 1}' -d  --cert admin_d885fd.crt  --key admin_d885fd.key --cafile ca.crt `
